### ISSUE TITLE
normalizeDatetimeValue ignore milliseconds

### ISSUE THREAD

**Issue by nnrudakov** [yiisoft/yii2#16938]
### What steps will reproduce the problem?
```php
Yii::$app->formatter->asDatetime('1539770576.2434', 'd.MM, H:mm:ss.A'); // or in PHP syntax, whatever
```

### What is the expected result?
Date with milliseconds.

### What do you get instead?
Date without milliseconds because of [that](https://github.com/yiisoft/yii2/blob/master/framework/i18n/Formatter.php#L830). Value becomes as integer.

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.15.1
| PHP version      | 7.2
| Operating system | Ubuntu

**Comment by rrmontuan** [comment]
I'm not sure if this is really a bug. I'm saying that beacuse even when we try the following code, creating a Datetime object, we don't get the milliseconds:

`print_r(new DateTime('@1539770576.2434'));`

The only way that I've got the desired result was passing a Datetime object as parameter and using a php datetime format, like this:

`$now = DateTime::createFromFormat('U.u', microtime(true));
Yii::$app->formatter->asDatetime($now, 'php:m-d-Y H:i:s.u');`

Besides that, using the Yii notation for formatting won't work because if you do that, Yii will try to convert ICU format to PHP Format (as you can see here  [here](https://github.com/yiisoft/yii2/blob/master/framework/i18n/Formatter.php#L781)) and the method **convertDateIcuToPhp** that comes originally from the **BaseFormatConverter** is replacing the A (representing milliseconds) for '' (empty) [here](https://github.com/yiisoft/yii2/blob/master/framework/helpers/BaseFormatConverter.php#L205).

**Comment by nnrudakov** [comment]
Maybe it's not a bug from developers point of view. Maybe `u` was skipped for some important development reasons. But if Yii docs says
> You can specify date and time formats using the ICU syntax. You can also use the PHP date() syntax
I expect date will be exactly in that format. I think current behavior is incorrect.

So, if I have to create `DateTime` object, I don't need `formatter` at all. I can do anything with dates by `DateTime`.

**Comment by rrmontuan** [comment]
@nnrudakov yeah, what you've said makes sense. We could change the Formatter class, veryfing if the parameter it's a timestamp with milliseconds. If so, we can use the method **DateTime::createFromFormat** to create the object. We also need to convert **u** (milliseconds) to **A** when format is passed using the PHP notation.

What do you think about it?

**Comment by rrmontuan** [comment]
I' tried to solve this problem but it seems that the format method (from Datetime class) and the format method (from IntlDateFormatter class), they produce different outputs (pretty similar, but not equal. The precision is also different). So, if someone is using the **Intl Extension** will get one result, and another result if it's not using the extension.  I've tried the following code:

```
$timestamp = DateTime::createFromFormat("U.u", 1544533172.478800, new DateTimeZone("UTC"));
$formatter = new IntlDateFormatter("en-US", IntlDateFormatter::NONE, IntlDateFormatter::NONE, 'UTC', null, "yyyy-MM-dd HH:mm:ss.A");
$datetime1 = $formatter->format($timestamp);

$timestamp2 = DateTime::createFromFormat("U.u", 1544533172.478800, new DateTimeZone("UTC"));
$datetime2 = $timestamp2->format("Y-m-d h:i:s.u");

echo "Datetime 1 : " . $datetime1 . "<br />";
echo "Datetime 2 : " . $datetime2;
```

Any sugestions?
